From c488bab801f4342ddc1599ec1cd821182da05aab Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <emil.renner.berthing@canonical.com>
Date: Sat, 12 Oct 2024 22:00:57 +0200
Subject: [PATCH 45/47] riscv: dts: thead: Add TH1520 AON and AP reset
 controllers

Add device tree nodes for the AON and AP reset controllers on the
T-Head TH1520 SoC.

Signed-off-by: Emil Renner Berthing <emil.renner.berthing@canonical.com>
---
 arch/riscv/boot/dts/thead/th1520.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/riscv/boot/dts/thead/th1520.dtsi b/arch/riscv/boot/dts/thead/th1520.dtsi
index cb14bdcbe170..85520ca700b8 100644
--- a/arch/riscv/boot/dts/thead/th1520.dtsi
+++ b/arch/riscv/boot/dts/thead/th1520.dtsi
@@ -653,6 +653,12 @@ i2c5: i2c@fff7f2c000 {
 			status = "disabled";
 		};
 
+		aprst: reset-controller@ffff015000 {
+			compatible = "thead,th1520-reset-ap";
+			reg = <0xff 0xff015000 0x0 0x2000>;
+			#reset-cells = <1>;
+		};
+
 		timer4: timer@ffffc33000 {
 			compatible = "snps,dw-apb-timer";
 			reg = <0xff 0xffc33000 0x0 0x14>;
@@ -724,6 +730,12 @@ aogpio: gpio-controller@0 {
 			};
 		};
 
+		aonrst: reset-controller@fffff44000 {
+			compatible = "thead,th1520-reset-aon";
+			reg = <0xff 0xfff44000 0x0 0x2000>;
+			#reset-cells = <1>;
+		};
+
 		padctrl_aosys: pinctrl@fffff4a000 {
 			compatible = "thead,th1520-pinctrl";
 			reg = <0xff 0xfff4a000 0x0 0x2000>;
-- 
2.47.0

